### ISSUE TITLE
Basic matchmaking

### ISSUE THREAD

**Issue by borrascador** [borrascador/rain#298]
Initial features:
- Allow players to start their own parties and to join other parties.
- For now all parties will be public
- Party capacity will be 5
- Recycle story interface

Future features in #302 
- Private parties
- Custom party capacity
- Custom HTML user interface
- Additional data fields

Since tribes are individual  the flow for a player without a party logging in could be:

> login/register -> choose tribe -> choose/create party -> play

The flow for a player with an existing party will be:

> login -> play

**Comment by amos1814** [comment]
Estimate: 2-4 hours (after #299 is complete)

**Comment by amos1814** [comment]
Each object in party
party: []
will have new fields 
NPC: boolean
self: boolean 
name: String

If someone joins your party or logs in, NPC would be set to false. If someone logs out, NPC for that character would be set to true.

I think we should have some visual distinction between self, NPC, and PC. Once a player joins your party, their name would be attached to the party member whether that player is online or not. This name can be displayed (maybe hover text with more information about that player). Also it would be nice if PCs could be prevented from being selected for movement.

**Comment by amos1814** [comment]
UPDATE message when player joins your party

**Comment by borrascador** [comment]
I think I will work on chat before this

**Comment by borrascador** [comment]
When I receive the login response with party members, I get duplicates of NPC 1 and NPC 2, (but no duplicate of currently logged in player), and also the NPC characters don't have a `self` property

**Comment by amos1814** [comment]
I think I solved the duplicate issue
